### PR TITLE
Use convert_element_type instead of device_put_raw.

### PR DESCRIPTION
**Pull request by pschuh**

### CODE REVIEW

**Comment by jakevdp** [comment]
Hi - out of curiosity, what was the impetus for this change? I ask because this kind of change tends to have subtle unintented consequences, so I'd prefer to avoid it unless there is good reason.

**Comment by mattjj** [comment]
Also, this code was written this way for performance, so we'd want to check some benchmarks for any change! #3350 has some basic microbenchmarks for `jnp.array`. (Now that we have some basic benchmark infrastructure, we should add this kind of stuff! )

File: jax/_src/numpy/lax_numpy.py [hunk 1]
```diff
@@ -2944,20 +2944,20 @@ def array(object, dtype=None, copy=True, order="K", ndmin=0):
   if type(object) is np.ndarray:
     _inferred_dtype = object.dtype and dtypes.canonicalize_dtype(object.dtype)
     lax._check_user_dtype_supported(_inferred_dtype, "array")
-    out = _device_put_raw(object, weak_type=weak_type)
+    out = np.array(object, copy=copy, dtype=dtype)
```
Comment (pschuh) [inline jax/_src/numpy/lax_numpy.py:2947]: I think this should maybe be _np_array ?

**Review by mattjj — APPROVED** [review]
Great find, Parker! (See jax-dev chat thread for more discussion.)

**Comment by mattjj** [comment]
I ran the simple `timeit jnp.array([0] * int(1e6))` benchmark from #3350 and there was no change (`214 ms ± 1.95` on master, `213 ms ± 1.88` on this branch).

**Comment by mattjj** [comment]
@jakevdp the main motivation is to avoid host-to-device (H2D) transfers during jit tracing, very much analogous to #6014. See the jax-dev chat thread for more discussion.

**Comment by mattjj** [comment]
@googlebot I'm a Googler, and I consent!

**Pull request merged**